src/bin/blobstore-s3/main.rs

```diff
@@ -8,7 +9,7 @@ async fn main() -> anyhow::Result<()> {
     let fut = run_provider_handler(provider.clone(), "blobstore-s3-provider")
```

Comment:
One more improvement might be to make it so callers don’t have to do this step and the serve functions just set it up for you